portfolio/src/main/webapp/script.js

```diff
@@ -30,26 +30,30 @@ function showSlides(n) {
   slides[slideIndex-1].style.display = "block";
 }
 
-function getRandomQuoteUsingArrowFunctions() {
-  fetch('/random-quote').then(response => response.text()).then((quote) => {
-    document.getElementById('quote-container').innerText = quote;
-  });
-}
-
 /**
- * Fetches a message from the server and adds it to the page.
+ * Fetches messages from the servers and adds them to the DOM.
  */
-function getMessage() {
-    fetch('/data').then(response => response.text()).then((message) => {
-        document.getElementById('msg-container').innerText = message;
-    });
+function getMessages() {
+  fetch('/data').then(response => response.json()).then((messages) => {
+    // messages is an array, not a string, so we have to
+    // reference its indexes to create HTML content
+    const messageListElement = document.getElementById('msg-container');
+    if (messages.length == 0) messageListElement.innerHTML = 'no messages';
+    else {
+        console.log("printing messages");
+        messageListElement.innerHTML = '';
+        var x;
+        for (x = 0; x < messages.length; x++) {
+            messageListElement.appendChild(
+            createListElement('Message '+ (x+1) + ': ' + messages[x]));
+        }
```

Comment:
```
for (let x in messages) {
  messageListElement.appendChild(
    createListElement('Message '+ (x + 1) + ': ' + messages[x]));
}
```
`let` vs `var` is a matter of scope for the variable: `let` is block scoped and `var` is function scoped. So prefer `let`.